azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/middleware/repository/MessageRepositoryMiddleware.kt

```diff
@@ -108,18 +115,29 @@ internal class MessageRepositoryMiddlewareImpl(
         notifyUpdate(dispatch)
     }
 
-    private fun processParticipantsRemoved(action: ParticipantAction.ParticipantsRemoved, dispatch: Dispatch) {
-        messageRepository.addLocalMessage(
-            MessageInfoModel(
-                id = "${messageRepository.getLastMessage()?.id?.toLong() ?: 0 + 1}",
-                participants = action.participants.map { it.displayName ?: "" },
-                content = null,
-                createdOn = OffsetDateTime.now(),
-                senderDisplayName = null,
-                messageType = ChatMessageType.PARTICIPANT_REMOVED
+    private fun processParticipantsRemoved(
+        action: ParticipantAction.ParticipantsRemoved,
+        dispatch: Dispatch,
+        localParticipant: LocalParticipantInfoModel
+    ) {
+        val isLocalParticipantEvicted =
+            action.participants.count { it.userIdentifier.id == localParticipant.userIdentifier } !=
+                action.participants.count()
+        if (isLocalParticipantEvicted) {
+            dispatch(ParticipantAction.LocalUserRemoved())
```

Comment:
we might also want to dispatch a system message action for the UI: 

<img width="359" alt="image" src="https://user-images.githubusercontent.com/109105353/202028294-149691d0-906b-4c66-a851-f0ce5a781959.png">

Reply:
done